From: SeongJae Park <sj@kernel.org>
Date: Mon, 12 May 2025 10:38:24 -0700
Subject: [PATCH] ==== set CONFIG_DAMON by default ====

mm/damon: build-enable essential DAMON components by default

As of this writing, multiple major distros including Alma, Amazon,
Android, CentOS, Debian, Fedora, and Oracle are build-enabling DAMON
(set CONFIG_DAMON[1]).  Configuring DAMON is not very easy, since it is
disabled by default, and there are multiple essential options that need
to be manually turned on, one by one.  Make it easier, by grouping
essential configurations to be enabled with one selection, and enabling
build of the essential parts of DAMON by default.

Note that build-enabling DAMON does not introduce any real risk, since
it makes no behavioral change by default.  It requires explicit user
requests to do anything.  Only one potential risk is making the size of
the kernel a little bit larger.  On a production-purpose configuration,
it increases the resulting kernel package binary size by about 0.1 % of
the final package file.  I believe that's too small to be a real problem
in common setups.

Changes from RFC
(https://lore.kernel.org/20250512182716.50245-1-sj@kernel.org)
- Rebase to latest mm-new
- Wordsmith cover letter and commit messages

[1] https://oracle.github.io/kconfigs/?config=UTS_RELEASE&config=DAMON

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_xph9uz8j | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_xph9uz8j

diff --git a/hkml_cv_bogus/hkml_cv_bogus_xph9uz8j b/hkml_cv_bogus/hkml_cv_bogus_xph9uz8j
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

